File: Wms.Api/Dto/Invoice/InvoiceDetails/InvoiceDetailsDto.cs
using System.Collections.Generic;

namespace Wms.Api.Dto.Invoice.InvoiceDetails
{
    public class InvoiceDetailsDto
    {
        public Guid Id { get; set; }
        public string Number { get; set; } = default!;
        public Guid? CustomerId { get; set; }
        public string? CustomerName { get; set; }
        public DateTime DateOfSale { get; set; }
        public string SalesPersonId { get; set; } = default!;
        public string? SalesPersonName { get; set; }
        public string? EOrderNumber { get; set; }
        public Guid? SalesTypeId { get; set; }
        public string? SalesTypeName { get; set; }
        public Guid? PaymentTypeId { get; set; }
        public string? PaymentTypeName { get; set; }
        public string? PaymentReference { get; set; }
        public string? Remark { get; set; }
        public decimal GrandTotal { get; set; }

        public ICollection<InvoiceItemDetailsDto> InvoiceItems { get; set; } = new List<InvoiceItemDetailsDto>();
    }
}

File: Frontend/src/interfaces/v12/invoice/invoiceDetailsDto.ts
import { guid } from "types/guid";
import { InvoiceItemDetailsDto } from "./invoiceItemDetailsDto";

export interface InvoiceDetailsDto {
  id: guid;
  number: string;
  customerId?: guid | null;
  customerName?: string | null;
  dateOfSale: string;
  salesPersonId: string;
  salesPersonName?: string | null;
  salesTypeId?: guid | null;
  salesTypeName?: string | null;
  paymentTypeId?: guid | null;
  paymentTypeName?: string | null;
  paymentReference?: string | null;
  remark?: string | null;
  grandTotal: number;
  warehouseId: guid;
  warehouseLabel?: string | null;
  createdAt?: string;
  createdById?: string;
  changedAt?: string | null;
  changedById?: string | null;
  invoiceItems: InvoiceItemDetailsDto[];
}
